client/src/App.js

```diff
@@ -18,30 +20,34 @@ import Followers from './components/Layout/Followers';
 import Followings from './components/Layout/Followings';
 import Unfollow from './components/Layout/Unfollow';
 import Footer from './components/Layout/Footer';
+import SinglePost from "./components/Layout/SinglePost";
 
 class App extends Component {
   render() {
     return (
-      <Router>
-        <div className="App">
-          <Navbar />
-          <div className="container">
-            <Route exact path="/" component={Landing} />
-            <Route exact path='/signup' component={Signup} />
-            <Route exact path="/login" component={Login} />
-            <Route exact path="/home" component={Homepage} />
-            <Route exact path="/suggestion" component={Suggestion} />
-            <Route exact path="/edit" component={EditProfile} />
-            <Route exact path='/changepassword' component={ChangePassword} />
-            <Route exact path="/profile" component={Profile} />
-            <Route path='/create' component={CreatePost} />
-            <Route exact path='/followers' component={Followers} />
-            <Route exact path='/followings' component={Followings} />
-            <Route exact path='/unfollow' component={Unfollow} />
+      <Provider store={store}>
+        <Router>
+          <div className="App">
+            <Navbar />
+            <div className="container">
+              <Route exact path="/" component={Landing} />
+              <Route exact path='/signup' component={Signup} />
+              <Route exact path="/login" component={Login} />
+              <Route exact path="/home" component={Homepage} />
+              <Route exact path="/suggestion" component={Suggestion} />
+              <Route exact path="/edit" component={EditProfile} />
+              <Route exact path='/changepassword' component={ChangePassword} />
+              <Route exact path="/profile" component={Profile} />
+              <Route exact path="/single-post" component={SinglePost} />
+              <Route path='/create' component={CreatePost} />
+              <Route exact path='/followers' component={Followers} />
+              <Route exact path='/followings' component={Followings} />
```

Comment:
U might get Merge conflicts if so then Keep the Followers , following and Unfollow /path as from master